contrib/envoyproxy/envoy/cmd/serviceextensions/main.go

```diff
@@ -0,0 +1,134 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package main
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"os"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/envoyproxy/envoy"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+
+	extproc "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	"github.com/gorilla/mux"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/reflection"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// AppsecCalloutExtensionService defines the struct that follows the ExternalProcessorServer interface.
+type AppsecCalloutExtensionService struct {
+	extproc.ExternalProcessorServer
+}
+
+type serviceExtensionConfig struct {
+	extensionPort   string
+	extensionHost   string
+	healthcheckPort string
+}
+
+func loadConfig() serviceExtensionConfig {
+	extensionPort := os.Getenv("DD_SERVICE_EXTENSION_PORT")
+	if extensionPort == "" {
+		extensionPort = "443"
+	}
+
+	extensionHost := os.Getenv("DD_SERVICE_EXTENSION_HOST")
+	if extensionHost == "" {
+		extensionHost = "0.0.0.0"
+	}
+
+	healthcheckPort := os.Getenv("DD_SERVICE_EXTENSION_HEALTHCHECK_PORT")
+	if healthcheckPort == "" {
+		healthcheckPort = "80"
+	}
+
+	return serviceExtensionConfig{
+		extensionPort:   extensionPort,
+		extensionHost:   extensionHost,
+		healthcheckPort: healthcheckPort,
+	}
+}
+
+func main() {
+	var extensionService AppsecCalloutExtensionService
+
+	// Force set ASM as enabled only if the environment variable is not set
+	// Note: If the environment variable is set to false, it should be disabled
+	if os.Getenv("DD_APPSEC_ENABLED") == "" {
+		if err := os.Setenv("DD_APPSEC_ENABLED", "1"); err != nil {
+			log.Error("service_extension: failed to set DD_APPSEC_ENABLED environment variable: %v\n", err)
+		}
+	}
```

Comment:
Starting from yesterday lol we have a tracer.Start option to force appsec start

Reply:
Applied c1187be 😄